src/query/functions/src/aggregates/aggregator_common.rs

```diff
@@ -154,14 +156,14 @@ pub fn eval_aggr(
 }
 
 #[inline]
-pub fn serialize_state<W: std::io::Write, T: serde::Serialize>(
+pub fn borsh_serialize_state<W: std::io::Write, T: BorshSerialize>(
     writer: &mut W,
     value: &T,
 ) -> Result<()> {
-    bincode_serialize_into_buf(writer, value)
+    borsh_serialize_into_buf(writer, value)
 }
 
 #[inline]
-pub fn deserialize_state<T: serde::de::DeserializeOwned>(slice: &mut &[u8]) -> Result<T> {
-    bincode_deserialize_from_stream(slice)
+pub fn borsh_deserialize_state<T: BorshDeserialize>(slice: &mut &[u8]) -> Result<T> {
+    borsh_deserialize_from_slice(slice)
```

Comment:
borsh_deserialize_from_slice will set the new position of the slice after read once? bincode_deserialize_from_stream does.

Reply:
nop, I realize this, currently borsh not provide a stream decode, we need fork it and impl one.